proofs/alf/cvc5/programs/Strings.smt3

```diff
@@ -325,7 +325,7 @@
 )
 
 ; Compute the eager reduction predicate for (str.code s)
-(define-fun string_eager_reduction_to_code ((s String)) Bool
+(define string_eager_reduction_to_code ((s String))
```

Comment:
I would argue we shouldn't do this in the "production" version of the signature, since `:type` only serves as a sanity check.  You can always use documentation instead of requiring that the checker invoke type checking here.

We could also have a "debug" version of the signature where `:type` is added wherever applicable. Note this impacts performance.